src/Options.cpp

```diff
@@ -4276,6 +4270,8 @@ COMPAS_VARIABLE Options::OptionValue(const T_ANY_PROPERTY p_Property) const {
         case PROGRAM_OPTION::PPI_LOWER_LIMIT                                : value = PulsationalPairInstabilityLowerLimit();                               break;
         case PROGRAM_OPTION::PPI_UPPER_LIMIT                                : value = PulsationalPairInstabilityUpperLimit();                               break;
 
+        case PROGRAM_OPTION::QCRIT_PRESCRIPTION                             : value = static_cast<int>(QCritPrescription());                                break;
```

Comment:
I'll defer to the wisdom of @jeffriley to give you a more enlightened answer. I just copied the formatting from one of the other prescriptions.

Reply:
@avigna @reinhold-willcox 

@avigna I will assume you want to know about 'static_cast' (vs any other type of cast) rather than assume you mistake this for a static function call...

This function returns the values of program options so they can be printed/logged to the log files.  In this case, the program option is an 'enum class':

enum class QCRIT_PRESCRIPTION: int { NONE, CLAEYS };

Let's say we remove the comments, and make it

enum class QCRIT_PRESCRIPTION: int { NONE, CLAEYS, GE15, GE15_IC};

We use enum classes to help prevent developers accidentally using numbers outside the range of values permitted for the variable.  For example, in this case, the valid qcrit prescriptions are NONE and CLAEYS, GE15, and GE15_IC.  The way it is declared, the integer values are NONE = 0, CLAEYS = 1, GE15 = 2, and GE15_IC = 3.  Our code (presumably) does different things for those values - bit only those values.  We don't want someone inadvertently assigning, say, 15 to the qcrit prescription - we don't know what that is.  So that's why we use enum classes - it restricts the values we can assign to a variable of that type and hopefully helps prevent inadvertent errors.

So that's the background.  

When we want to print the value of the option to a log file, we want to print the integer value - so we need to cast the variable to an int.  We could just use

value = (int)QCritPrescription();

instead of

value = static_cast<int>(QCritPrescription());

and get the same result.  In this case...  The classic C cast '(int)QCritPrescription()' does no checking - it just converts the value to an int.  'static_cast' on the other hand does some checking to ensure that the variable being cast to an int can actually be cast to an int with no errors.  In this case we know it can, but it is good practice to use 'static_int' over '(int)' - not in all cases, but in most (as always, there are exceptions, and pragmatism should prevail over dogma).  'static_cast' assumes that there is a valid conversion in the language, or an appropriate constructor that makes the cast possible (assuming static typing).

There are other types of cast in C++: 'dynamic_cast', 'const_cast', and 'reinterpret_cast'. Some of those are dangerous (in that if you don't know for sure what you're doing you'll end up causing problems), and they don't apply here anyway (they could, but static_cast is the best choice).  Google them if you really want to know about them - the discussion here would be way too long if I described them...

Reply:
Oh wow Jeff, this was a great explanation! Thanks :)